mailpoet/lib/Automation/Engine/Control/StepHandler.php

```diff
@@ -211,4 +213,33 @@ private function getSubjectEntries(WorkflowRun $workflowRun, array $requiredSubj
     }
     return $subjectEntries;
   }
+
+  private function postProcessWorkflowRun(int $workflowRunId): void {
+    $workflowRun = $this->workflowRunStorage->getWorkflowRun($workflowRunId);
+    if (!$workflowRun) {
+      return;
+    }
+    $workflow = $this->workflowStorage->getWorkflow($workflowRun->getWorkflowId());
+    if (!$workflow) {
+      return;
+    }
+    $this->postProcessWorkflow($workflow);
+  }
+
+  private function postProcessWorkflow(Workflow $workflow): void {
+    if ($workflow->getStatus() === Workflow::STATUS_DEACTIVATING) {
+      $activeRuns = array_filter(
+        $this->workflowRunStorage->getWorkflowRunsForWorkflow($workflow),
```

Comment:
I had only a quick look, but since you'll probably be starting earlier tomorrow I'll leave a note here that I noticed this could be a performance bottleneck over time. This would load all (including historical) workflows.

Could we instead do something like `getCountForWorkflow(Workflow $workflow, string $status)`? That could be much more efficient.

Reply:
Thats true. Addressed in [a494061](https://github.com/mailpoet/mailpoet/pull/4452/commits/a494061622c7a7f4037cd8ab29ac811c6b4d01fe) and [66337b7](https://github.com/mailpoet/mailpoet/pull/4452/commits/66337b7208ac8170472c1385194850956b9dbb5b)